### DISCUSSION TITLE
.env alternative?

### DISCUSSION THREAD

**Discussion by Jarco-dev** [prisma/prisma#10477] [Q&A] [answered]
Is there anyway i can import the database url in to schema.prisma with preferably a .js file 
or at the very least atleast change the location of the .env  (i highly prefer js for my config files as that means i can use comments)

**Comment by austincrim** [comment, marked as answer]
Hey @Jarco-dev 👋,

You can pass in the database connection string when you instantiate the `PrismaClient`:
```ts
import dbUrl from './some-file.js'
import { PrismaClient} from '@prisma/client'

const prisma = new PrismaClient({
  datasources: {
    db: {
      url: dbUrl
    },
  },
})
```

As an aside, you can put comments in `.env` files like this:
```.env
# This is a comment!
DATABASE_URL="some://url"
```

Let me know if you have questions!

**Reply by Jarco-dev** [reply]
I was aware that you can enter a database url when creating the prisma client, but I don't think the cli will be able to access the database url for migrations can it?

**Reply by janpio** [reply]
No, but you can do whatever you want to put the value inside the actual environment variable. The `.env` file is just an included, by default support way. You can also set the variable value manually, or execute any other script before running a `prisma migrate` command for example.

**Reply by Jarco-dev** [reply]
I don't really get how else to do it by manually do you mean for example providing it as an argument or how else?

**Reply by janpio** [reply]
You set them in the environment of the computer where you are running Prisma. On Linux or macOS you can execute `export DATABASE_URL="foo"` before you run the CLI command for example.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
